pkg/scheduler/plugins/gang/gang.go

```diff
@@ -144,8 +145,10 @@ func (gp *gangPlugin) OnSessionOpen(ssn *framework.Session) {
 
 func (gp *gangPlugin) OnSessionClose(ssn *framework.Session) {
 	for _, job := range ssn.Jobs {
-		if len(job.TaskStatusIndex[api.Allocated]) != 0 {
-			ssn.Backoff(job, arbcorev1.UnschedulableEvent, "not enough resource for job")
+		if len(job.TaskStatusIndex[api.Pending]) != 0 {
```

Comment:
When `Gang-scheduling` did not meet, we need to record an event at job level; if `gang-scheduling` does not meet, the task is allocated but not bind to the host.

Reply:
btw, we need to support `totalTasksNum > minMemberNum` case for elastic jobs, e.g. Spark, in future.

Reply:
> When `Gang-scheduling` did not meet, we need to record an event at job level; if `gang-scheduling` does not meet, the task is allocated but not bind to the host.

Not sure what you mean here. There may be gang jobs that failed scheduling with no `Allocated` tasks (all `Pending`), right ?

Reply:
> btw, we need to support `totalTasksNum > minMemberNum` case for elastic jobs, e.g. Spark, in future.

When calculating the total number of task, the size of JobInfo->Tasks is used, which should cover the case you mention, correct ?